modules/images/dominant-color/class-dominant-color-image-editor-gd.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * WordPress Image Editor Class for Image Manipulation through GD
+ * with dominant color detection
+ *
+ * @package performance-lab
+ * @group dominant-color
+ *
+ * @since n.e.x.t
+ */
+
+/**
+ * WordPress Image Editor Class for Image Manipulation through GD
+ * with dominant color detection
+ *
+ * @since n.e.x.t
+ *
+ * @see WP_Image_Editor
+ */
+class Dominant_Color_Image_Editor_GD extends WP_Image_Editor_GD {
+
+	/**
+	 * Get dominant color from a file.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $default_color Optional. Hex color string, without leading #. Default is light grey.
+	 * @return string Dominant hex color string.
+	 */
+	public function get_dominant_color( $default_color = 'eee' ) {
```

Comment:
Personally I would move this parameter into a method where we can introduce a filter or the default color can be override.

For example:

```php
public function get_default_color() {
  return apply_filters( 'filter_name', 'ccc' );
} 
```

```suggestion
	public function get_dominant_color() {
		if ( ! $this->image ) {

			return $this->get_default_color();
		}	
```

Reply:
I am going to push back on this approach and add the filter in the calling function dominant_color_get()